contrib/babelfishpg_tsql/src/linked_servers.c

```diff
@@ -88,6 +88,33 @@ linked_server_msg_handler(LinkedServerProcess lsproc, int error_code, int state,
 	return 0;
 }
 
+/*
+ * Handle any error encountered in TDS client library itself
+ */
+static int
+linked_server_err_handler(LinkedServerProcess lsproc, int severity, int db_error, int os_error, char *db_err_str, char *os_err_str)
+{
+	StringInfoData buf;
+
+	initStringInfo(&buf);
+
+	appendStringInfo(
+		&buf,
+		"TDS client library error: DB #: %i, DB Msg: %s, OS #: %i, OS Msg: %s, Level: %i",
+		db_error,
+		db_err_str ? db_err_str : "",
+		os_error,
+		os_err_str ? os_err_str : "",
+		severity
+	);
```

Comment:
we should construct error message in a way that includes key: value thing only if value is valid. Otherwise there is not point of adding empty string.

Reply:
I kept it deliberately like this because this is how it gets handled in tds_fdw as well.

Reply:
https://github.com/tds-fdw/tds_fdw/blob/c24be29f5f4a9c5ba87b8e025fc898637e589583/src/tds_fdw.c#L3937)